onnxruntime/core/providers/cuda/reduction/reduction_functions.cu

```diff
@@ -347,9 +345,6 @@ __global__ void reduce_matrix_rows_kernel(const TIn* input, TOut* output, int m,
     if (threadIdx.y == 0) {
       atomic_add(output + col, TOut(shared_memory[threadIdx.x]));
     }
-
-    // Make sure all values in shared memory have been written into the output memory.
-    __syncthreads();
   }
```

Comment:
Why do we remove this line?

Reply:
The barrier (__syncthreads()) is not needed here since there is no co-operation between threads after this point and as you pointed out, there is another barrier at line 333.

Reply:
Makes sense. My comment is not blocking now.

Reply:
Thanks for reviewing!!